### PR TITLE
feat: Update `karpenter` example to use the EKS module, latest version of Karpenter, and run on Fargate

### PR DESCRIPTION
**Pull request by bryantbiggs**
### What does this PR do?

- Update the `karpenter` example to:
	- Use the `terraform-aws-eks` module for cluster creation
	- Use the latest Karpenter settings and recommendations (i.e. - `AWSNodeTemplate`, consolidation, native spot termination handling, etc.)
	- Run Karpenter on EKS Fargate
- Update the `karpenter` addon to allow passing in a computed SQS ARN value without error and use the latest Karpenter release version
- Remove Datadog operator addon from example - requires an API key
 
### Motivation

- Updates the example blueprint to the current practices used when running Karpenter on EKS

### More

- [x] Yes, I have tested the PR using my local account setup (Provide any test evidence report under Additional Notes)
- [ ] Yes, I have added a new example under [examples](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/examples) to support my PR
- [ ] Yes, I have created another PR for add-ons under [add-ons](https://github.com/aws-samples/eks-blueprints-add-ons) repo (if applicable)
- [ ] Yes, I have updated the [docs](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/docs) for this feature
- [x] Yes, I ran `pre-commit run -a` with this PR

**Note**: Not all the PRs require a new example and/or doc page. In general:
- Use an existing example when possible to demonstrate a new addons usage
- A new docs page under `docs/add-ons/*` is required for new a new addon

### For Moderators

- [ ] E2E Test successfully complete before merge?

### CODE REVIEW

File: examples/karpenter/main.tf [hunk 1]
```diff
@@ -2,30 +2,40 @@ provider "aws" {
...
   token                  = data.aws_eks_cluster_auth.this.token
 }
 
 data "aws_eks_cluster_auth" "this" {
-  name = module.eks_blueprints.eks_cluster_id
+  name = module.eks.cluster_name
+}
+
+data "aws_ecrpublic_authorization_token" "token" {
+  provider = aws.virginia
```
Comment (csantanapr) [inline examples/karpenter/main.tf:38]: TIL!

**Review by csantanapr — APPROVED** [review]
/lgtm

File: examples/karpenter/main.tf [hunk 2]
```diff
@@ -45,209 +53,231 @@ locals {
...
+################################################################################
+
+#tfsec:ignore:aws-eks-enable-control-plane-logging
+module "eks" {
+  source  = "terraform-aws-modules/eks/aws"
+  version = "~> 19.5"
+
+  cluster_name                   = local.name
+  cluster_version                = "1.24"
+  cluster_endpoint_public_access = true
```
Comment (FernandoMiguel) [inline examples/karpenter/main.tf:67]: this is a bad example default, from a security point of view, considering many will opy the examples as their deployment reference.

Comment (bryantbiggs) [reply examples/karpenter/main.tf:67]: all examples, minus the fully private example, have the public endpoint enabled. This was done intentionally to make it easier to get started with interacting with the clusters from the examples as well as for testing from the GitHub workflows. The only change here is that it is now explicitly clear that the public endpoint is being enabled 

Was there some value that was meant to be added by posting this comment?

**Pull request merged**